### PR TITLE
TiDB Integration: user experience improvements and 2.1.0 release

### PR DESCRIPTION
**Pull request by SabaPing**
### What does this PR do?

This PR includes several improvements for TiDB Integration:

* [Improvement] Update default metrics.
* [Improvement] Update the default dashboard.
* [Improvement] Expose the `metrics` mapper config to let users customize more metrics.
* [Improvement] Let users to customize metric namespace.
* [Improvement] Update readme.
* [Improvement] Add more tests.

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [ ] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### CODE REVIEW

**Comment by SabaPing** [comment]
Question here: 

I tried to remove the unused dashboard. But the validation CR reported `Error: Short name `overview` at `assets/dashboards` is not allowed to be modified. Please revert to original value.`.

How can I solve this error?

Thanks!

File: tidb/DEVELOPMENT.md [hunk 1]
```diff
@@ -2,7 +2,10 @@
 
 1. Set up the [Agent dev tool][1].
 
-2. Install the [Datadog Agent][2].
+> Make sure you have the latest version of `ddev` installed.
+> Check the version by running `ddev --version`.
+
+2.Install the [Datadog Agent][2].
```
Comment (jtappa) [inline tidb/DEVELOPMENT.md:8]:
```suggestion
2. Install the [Datadog Agent][2].
```

File: tidb/README.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ Connect [TiDB][1] cluster to Datadog in order to:
 > **Note:** 
 >
 > - TiDB 4.0+ is required for this integration. 
-> - Integration of TiDB Cloud with Datadog is not available now.
+> - For TiDB Cloud, please refer to [TiDB Cloud Integration][12].
```
Comment (jtappa) [inline tidb/README.md:14]:
```suggestion
> - For TiDB Cloud, refer to the [TiDB Cloud Integration][12].
```

File: tidb/README.md [hunk 3]
```diff
@@ -145,16 +143,15 @@ _Available for Agent versions >6.0_
 
 See [metadata.csv][6] for a list of metrics provided by this check.
 
+> It is possible to use the `metrics` config to collect more other metrics from a TiDB cluster.
```
Comment (jtappa) [inline tidb/README.md:146]:
```suggestion
> It is possible to use the `metrics` config to collect other metrics from a TiDB cluster.
```

**Review by jtappa — CHANGES_REQUESTED** [review]

**Review by jtappa — APPROVED** [review]

File: tidb/assets/service_checks.json [hunk 4]
```diff
@@ -1,18 +1 @@
-[
```
Comment (coignetp) [inline tidb/assets/service_checks.json:1]: Could you add it back? Even though it's not used it's still sent, so it should be documented

**Review by coignetp — CHANGES_REQUESTED** [review]

File: tidb/assets/dashboards/one-screen-overview.json [hunk 5]
```diff
@@ -1 +0,0 @@
...:[]}},{"id":7739111915948736,"definition":{"title":"AVG of Compile Duration","title_size":"16","title_align":"left","show_legend":true,"legend_layout":"auto","legend_columns":["avg","min","max","value","sum"],"time":{},"type":"timeseries","requests":[{"q":"per_second(avg:tidb_cluster.tidb.session_compile_duration_seconds.sum{type:wait} by {host}), per_second(avg:tidb_cluster.tidb.session_compile_duration_seconds.count{type:wait} by {host}), per_second(avg:tidb_cluster.tidb.session_compile_duration_seconds.sum{type:wait} by {host})/per_second(avg:tidb_cluster.tidb.session_compile_duration_seconds.count{type:wait} by {host})","style":{"palette":"dog_classic","line_type":"solid","line_width":"normal"},"display_type":"line"}],"yaxis":{"scale":"linear","label":"","include_zero":true,"min":"auto","max":"auto"},"markers":[]}}]}},{"id":4061043905521020,"definition":{"title":"TiKV","type":"group","layout_type":"ordered","widgets":[]}},{"id":469462288990948,"definition":{"title":"PD","type":"group","layout_type":"ordered","widgets":[]}}],"template_variables":[{"name":"cluster","default":"*","prefix":"cluster"}],"layout_type":"ordered","is_read_only":false,"notify_list":[],"id":"8ci-dz8-ugh"}
```
Comment (coignetp) [inline tidb/assets/dashboards/one-screen-overview.json:1]: Is this PR removing this dashboard?

Comment (SabaPing) [reply tidb/assets/dashboards/one-screen-overview.json:1]: Yes, I tried to remove it but CI checks failed.

Comment (SabaPing) [reply tidb/assets/dashboards/one-screen-overview.json:1]: I try to keep one dashboard, whose title is `TiDB Overview`, and remove others.

Comment (coignetp) [reply tidb/assets/dashboards/one-screen-overview.json:1]: We avoid deleting dashboards, do you have any additional context on why this dashboard should be deleted?

Comment (SabaPing) [reply tidb/assets/dashboards/one-screen-overview.json:1]: Previously, TiDB Integration provided two dashboards, a "cluster overview" dashboard and a supplement to the "cluster overview".

In this new version we have simplified the metrics. with new metrics, the dashboard can be optimized to one. If we can't delete the dashboard, could you give us a better solution? Thanks.

File: tidb/manifest.json [hunk 6]
```diff
@@ -29,8 +28,8 @@
       "spec": "assets/configuration/spec.yaml"
     },
     "dashboards": {
-      "overview": "assets/dashboards/one-screen-overview.json",
-      "TiDB Overview": "assets/dashboards/overview.json"
+      "TiDB Overview": "assets/dashboards/overview.json",
+      "overview": "assets/dashboards/overview.json"
```
Comment (coignetp) [inline tidb/manifest.json:32]:
```suggestion
      "TiDB Overview": "assets/dashboards/overview.json"
```
After that you can delete the dashboard file

**Review by coignetp — CHANGES_REQUESTED** [review]

File: tidb/assets/dashboards/one-screen-overview.json (hunk 5 above)
Comment (coignetp) [reply tidb/assets/dashboards/one-screen-overview.json:1]: It's possible to delete the dashboard, but it's a bit tricky on our side so I prefer to make sure we do that properly for good reasons 😄 
You can remove the dashboard following this https://github.com/DataDog/integrations-extras/pull/1094#discussion_r765553318

**Comment by ChristineTChen** [comment]
Hi @SabaPing ,

Have you had a chance to try @coignetp's suggestion?

**Comment by SabaPing** [comment]
Sorry for long time no update. I'll work on it today and tomorrow.

**Comment by SabaPing** [comment]
I followed @coignetp 's suggestion and removed unused dashboard.
I also: 
- updated the remaining dashboard. (This dashboard should be the final version for v2.x.x TiDB Integration)
- improved tests

Thanks a lot!

File: tidb/metadata.csv [hunk 7]
```diff
@@ -1,43 +1,13 @@
...
-tidb_cluster.tikv_engine_size_bytes,count,,,,Sum of engine size,1,tidb,
-tidb_cluster.tikv_channel_full_total,count,,,,Sum of channel full events,1,tidb,
-tidb_cluster.tikv_server_report_failure_msg_total,count,,,,Sum of server report failures,1,tidb,
-tidb_cluster.tikv_scheduler_context_total,count,,,,Sum of scheduler contexts,1,tidb,
-tidb_cluster.tikv_coprocessor_executor_count,count,,,,Sum of coprocessor executors,1,tidb,
-tidb_cluster.tikv_coprocessor_request_duration_seconds.sum,count,,,,Total number of coprocessor requests,1,tidb,
-tidb_cluster.tikv_coprocessor_request_duration_seconds.count,count,,,,Sum of coprocessor request duration,1,tidb,
-tidb_cluster.tidb_session_parse_duration_seconds.sum,count,,,,Total number of session parses,1,tidb,
-tidb_cluster.tidb_session_parse_duration_seconds.count,count,,,,Sum of session parse duration,1,tidb,
+tidb_cluster.tidb_server_handle_query_duration_seconds,count,,,,The distribution of TiDB query durations,1,tidb,
```
Comment (sarah-witt) [inline tidb/metadata.csv:5]:
```suggestion
tidb_cluster.tidb_server_handle_query_duration_seconds,count,,second,,The distribution of TiDB query durations,1,tidb,
```
It's not required, but could you add units for all of the metrics?

File: tidb/README.md [hunk 8]
```diff
@@ -145,16 +143,15 @@ _Available for Agent versions >6.0_
 
 See [metadata.csv][6] for a list of metrics provided by this check.
 
+> It is possible to use the `metrics` config to collect other metrics from a TiDB cluster.
```
Comment (sarah-witt) [inline tidb/README.md:146]:
```suggestion
> It is possible to use the `metrics` configuration option to collect additional metrics from a TiDB cluster.
```

**Review by sarah-witt — COMMENTED** [review]
Hi @SabaPing, thanks for the updates! I just left a few more comments

File: tidb/metadata.csv (hunk 7 above)
Comment (SabaPing) [reply tidb/metadata.csv]: `tidb_cluster.tidb_server_handle_query_duration_seconds` is actually `distribution` type.

How can I define `distribution` metric type? I got an error 

> "distribution" is an invalid metric_type 

when I tried to set type to distribution.

I also checked the validation source code and got this:
```python
VALID_METRIC_TYPE = {'count', 'gauge', 'rate'}
```

Comment (sarah-witt) [reply tidb/metadata.csv]: Can you revert back to specifying the two metrics you defined previously? These count and sum metrics are sent by the check for the distribution 
```
tidb_cluster.tidb_server_handle_query_duration_seconds.count,count,,,,The total number of handled queries in server,1,tidb,
tidb_cluster.tidb_server_handle_query_duration_seconds.sum,count,,,,The sum of handled query duration in server,1,tidb,
```

**Review by sarah-witt — COMMENTED** [review]
Hi @SabaPing can you also update the PR with the most recent changes in master? This [change](https://github.com/DataDog/integrations-extras/pull/1111) will allow more tests to run on your code

**Review by sarah-witt — APPROVED** [review]

File: tidb/README.md [hunk 9]
```diff
@@ -181,3 +183,6 @@ Need help? Contact [Datadog support][7].
 [9]: https://docs.datadoghq.com/agent/kubernetes/log/
 [10]: https://docs.datadoghq.com/agent/guide/community-integrations-installation-with-docker-agent
 [11]: https://github.com/DataDog/integrations-extras/blob/master/tidb/assets/service_checks.json
+[12]: https://docs.datadoghq.com/integrations/tidb_cloud/
+[13]: https://docs.pingcap.com/tidb/stable/tiup-playground
+[14]:
```
Comment (sarah-witt) [inline tidb/README.md:188]: Can you add this link?

File: tidb/README.md [hunk 10]
```diff
@@ -158,6 +153,13 @@ See [service_checks.json][11] for a list of service checks provided by this inte
 
 ## Troubleshooting
 
+### Missing CPU and Memory metrics for TiKV and TiFlash instances on macOS
+
+You will not get CPU or Memory metrics for TiKV and TiFlash instances in two cases:
+
+- Running TiKV or TiFlash instances with [tiup playground][13] on macOS.
+- Or, running TiKV or TiFlash instances with [docker-compose up][14] on a new Apple M1 machine.
```
Comment (jtappa) [inline tidb/README.md:161]:
```suggestion
CPU and Memory metrics are not provided for TiKV and TiFlash instances in the following cases:

- Running TiKV or TiFlash instances with [tiup playground][13] on macOS.
- Running TiKV or TiFlash instances with [docker-compose up][14] on a new Apple M1 machine.
```

**Review by jtappa — CHANGES_REQUESTED** [review]

**Review by sarah-witt — APPROVED** [review]

File: tidb/manifest.json [hunk 11]
```diff
@@ -29,7 +28,6 @@
       "spec": "assets/configuration/spec.yaml"
     },
     "dashboards": {
-      "overview": "assets/dashboards/one-screen-overview.json",
       "TiDB Overview": "assets/dashboards/overview.json"
```
Comment (hithwen) [inline tidb/manifest.json:31]:
```suggestion
      "overview": "assets/dashboards/overview.json"
```
This key cannot be changed, but don't worry, the name in the actual overview.json file will be used

**Pull request merged**